File: app/src/main/java/com/muratcan/apps/petvaccinetracker/SettingsActivity.java
package com.muratcan.apps.petvaccinetracker;

import android.app.TimePickerDialog;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.appbar.MaterialToolbar;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.card.MaterialCardView;
import com.google.android.material.checkbox.MaterialCheckBox;
import com.muratcan.apps.petvaccinetracker.util.NotificationSettingsHelper;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class SettingsActivity extends AppCompatActivity {
    private NotificationSettingsHelper settingsHelper;
    private LinearLayout reminderDaysContainer;
    private TextView notificationTimeText;
    private MaterialButton saveButton;
    private List<MaterialCheckBox> reminderCheckBoxes;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        settingsHelper = new NotificationSettingsHelper(this);
        reminderCheckBoxes = new ArrayList<>();

        initViews();
        setupToolbar();
        loadCurrentSettings();
        setupClickListeners();
    }

    private void initViews() {
        reminderDaysContainer = findViewById(R.id.reminder_days_container);
        notificationTimeText = findViewById(R.id.notification_time_text);
        saveButton = findViewById(R.id.save_settings_button);
    }

    private void setupToolbar() {
        MaterialToolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setTitle("Notification Settings");
        }
    }

    private void loadCurrentSettings() {
        // Load reminder days
        setupReminderDaysCheckboxes();

        // Load notification time
        updateNotificationTimeDisplay();
    }

    private void setupReminderDaysCheckboxes() {
        reminderDaysContainer.removeAllViews();
        reminderCheckBoxes.clear();

        int[] availableDays = NotificationSettingsHelper.getAllAvailableReminderDays();
        int[] enabledDays = settingsHelper.getReminderDays();

        for (int day : availableDays) {
            MaterialCheckBox checkBox = new MaterialCheckBox(this);
            checkBox.setText(NotificationSettingsHelper.getReminderDayDisplayText(this, day));
            checkBox.setTag(day);

            // Check if this day is currently enabled
            boolean isEnabled = false;
            for (int enabledDay : enabledDays) {
                if (enabledDay == day) {
                    isEnabled = true;
                    break;
                }
            }
            checkBox.setChecked(isEnabled);

            reminderCheckBoxes.add(checkBox);
            reminderDaysContainer.addView(checkBox);
        }
    }

    private void updateNotificationTimeDisplay() {
        int hour = settingsHelper.getNotificationHour();
        int minute = settingsHelper.getNotificationMinute();

        String timeText = String.format(Locale.getDefault(), "%02d:%02d", hour, minute);
        notificationTimeText.setText("Remind me at: " + timeText);
    }

    private void setupClickListeners() {
        // Time picker
        findViewById(R.id.notification_time_container).setOnClickListener(v -> showTimePicker());

        // Save button
        saveButton.setOnClickListener(v -> saveSettings());
    }

    private void showTimePicker() {
        int currentHour = settingsHelper.getNotificationHour();
        int currentMinute = settingsHelper.getNotificationMinute();

        TimePickerDialog timePickerDialog = new TimePickerDialog(
                this,
                (view, hourOfDay, minute) -> {
                    settingsHelper.setNotificationTime(hourOfDay, minute);
                    updateNotificationTimeDisplay();
                },
                currentHour,
                currentMinute,
                true // 24-hour format
        );

        timePickerDialog.show();
    }

    private void saveSettings() {
        // Get selected reminder days
        List<Integer> selectedDays = new ArrayList<>();
        for (MaterialCheckBox checkBox : reminderCheckBoxes) {
            if (checkBox.isChecked()) {
                selectedDays.add((Integer) checkBox.getTag());
            }
        }

        if (selectedDays.isEmpty()) {
            Toast.makeText(this, "Please select at least one reminder option", Toast.LENGTH_SHORT).show();
            return;
        }

        // Convert to array and save
        int[] reminderDays = selectedDays.stream().mapToInt(Integer::intValue).toArray();
        settingsHelper.setReminderDays(reminderDays);

        Toast.makeText(this, "Settings saved successfully", Toast.LENGTH_SHORT).show();
        finish();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/muratcan/apps/petvaccinetracker/SettingsActivity.kt
package com.muratcan.apps.petvaccinetracker

import android.app.TimePickerDialog
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.appbar.MaterialToolbar
import com.google.android.material.button.MaterialButton
import com.google.android.material.checkbox.MaterialCheckBox
import com.muratcan.apps.petvaccinetracker.util.NotificationHelper
import com.muratcan.apps.petvaccinetracker.util.NotificationSettingsHelper
import java.util.Locale

class SettingsActivity : AppCompatActivity() {
    private lateinit var settingsHelper: NotificationSettingsHelper
    private lateinit var reminderDaysContainer: LinearLayout
    private lateinit var notificationTimeText: TextView
    private lateinit var saveButton: MaterialButton
    private val reminderCheckBoxes = mutableListOf<MaterialCheckBox>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        settingsHelper = NotificationSettingsHelper(this)

        initViews()
        setupToolbar()
        loadCurrentSettings()
        setupClickListeners()
    }

    private fun initViews() {
        reminderDaysContainer = findViewById(R.id.reminder_days_container)
        notificationTimeText = findViewById(R.id.notification_time_text)
        saveButton = findViewById(R.id.save_settings_button)
    }

    private fun setupToolbar() {
        val toolbar = findViewById<MaterialToolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.apply {
            setDisplayHomeAsUpEnabled(true)
            title = "Notification Settings"
        }
    }

    private fun loadCurrentSettings() {
        // Load reminder days
        setupReminderDaysCheckboxes()

        // Load notification time
        updateNotificationTimeDisplay()
    }

    private fun setupReminderDaysCheckboxes() {
        reminderDaysContainer.removeAllViews()
        reminderCheckBoxes.clear()

        val availableDays = NotificationSettingsHelper.getAllAvailableReminderDays()
        val enabledDays = settingsHelper.getReminderDays()

        for (day in availableDays) {
            val checkBox = MaterialCheckBox(this).apply {
                text =
                    NotificationSettingsHelper.getReminderDayDisplayText(this@SettingsActivity, day)
                tag = day
                isChecked = day in enabledDays
            }

            reminderCheckBoxes.add(checkBox)
            reminderDaysContainer.addView(checkBox)
        }
    }

    private fun updateNotificationTimeDisplay() {
        val hour = settingsHelper.getNotificationHour()
        val minute = settingsHelper.getNotificationMinute()

        val timeText = String.format(Locale.getDefault(), "%02d:%02d", hour, minute)
        notificationTimeText.text = "Remind me at: $timeText"
    }

    private fun setupClickListeners() {
        // Time picker
        findViewById<View>(R.id.notification_time_container).setOnClickListener { showTimePicker() }

        // Save button
        saveButton.setOnClickListener { saveSettings() }
    }

    private fun showTimePicker() {
        val currentHour = settingsHelper.getNotificationHour()
        val currentMinute = settingsHelper.getNotificationMinute()

        val timePickerDialog = TimePickerDialog(
            this,
            { _, hourOfDay, minute ->
                settingsHelper.setNotificationTime(hourOfDay, minute)
                updateNotificationTimeDisplay()

                // Reschedule all existing notifications with new time
                NotificationHelper.rescheduleAllExistingNotifications(this)

                Toast.makeText(
                    this,
                    "Notification time updated for all existing reminders",
                    Toast.LENGTH_SHORT
                ).show()
            },
            currentHour,
            currentMinute,
            true // 24-hour format
        )

        timePickerDialog.show()
    }

    private fun saveSettings() {
        // Get selected reminder days
        val selectedDays = reminderCheckBoxes
            .filter { it.isChecked }
            .map { it.tag as Int }

        if (selectedDays.isEmpty()) {
            Toast.makeText(this, "Please select at least one reminder option", Toast.LENGTH_SHORT)
                .show()
            return
        }

        // Convert to array and save
        val reminderDays = selectedDays.toIntArray()
        settingsHelper.setReminderDays(reminderDays)

        // Reschedule all existing notifications with new settings
        NotificationHelper.rescheduleAllExistingNotifications(this)

        Toast.makeText(this, "Settings saved and notifications updated", Toast.LENGTH_SHORT).show()
        finish()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                finish()
                true
            }

            else -> super.onOptionsItemSelected(item)
        }
    }
} 